text/0156-columnar-encoding.md

```diff
@@ -0,0 +1,1031 @@
+[Zoom on the chart](https://raw.githubusercontent.com/lquerel/oteps/main/text/img/0156_summary_time_spent.png)
+
+> In conclusion, these benchmarks demonstrate the interest of integrating a column-oriented telemetry data protocol to
+> optimize bandwidth and processing speed in a batch processing context.
+
+### Why Apache Arrow and How to Use It?
+
+[Apache Arrow](https://arrow.apache.org/) is a versatile columnar format for flat and hierarchical data, well
+established in the industry. Arrow is optimized for:
+
+* column-oriented data exchange based on an in-memory format common among implementations, regardless of the language.
+The use of a serialization and deserialization mechanism is thus eliminated, allowing the Zero-copy.
+* in-memory analytic operations using modern hardware optimizations (e.g. SIMD)
+* integration with a large ecosystem (e.g. data pipelines, databases, stream processing, etc.)
+* language-independent
+
+All these properties make Arrow a great choice for a general-purpose telemetry protocol. Efficient implementations of
+Apache Arrow exist for most of the languages (Java, Go, C++, Rust, ...). Connectors with Apache Arrow buffer exist for
+well-known file format (e.g. Parquet) and for well-known backend (e.g. BigQuery).
+By reusing this existing infrastructure (see [Arrow ecosystem](https://arrow.apache.org/powered_by/)), we are accelerating the development and
+integration of the OpenTelemetry protocol while expanding its scope of application.
+
+Adapting the OTLP data format to the Arrow world (see below) is only part of the problem this proposal aims to describe.
+Many other design choices and trade-offs have been made, such as:
+
+- the organization of the data (i.e. schema and sorting) and the selection of the compression algorithm to optimize the compression ratio.
+- the way to serialize the Arrow data, the metadata, the dictionaries, and the selection of the transfer mode (reply/reply vs. bi-dir stream).
+- optimization of many parameters introduced in the system.
+
+![In-memory Apache Arrow RecordBatch](img/0156_OTEL%20-%20HowToUseArrow.png)
+
+### Integration Strategy and Phasing
+
+This OTEP enhances the existing OTel eco-system with an additional representation of telemetry data in columnar form to
+better support certain scenarios (e.g. cost-effectiveness to transmit large batch, multivariate time-series, advanced
+data processing, data minimization). All existing components will continue to be compatible and operational.
+
+A two-phase integration is proposed to allow incremental benefits.
+
+#### Phase 1
+
+This proposal is designed as a protocol extension compatible with the existing OTLP protocol. As illustrated in the
+following diagram, a new OTel Arrow to OTLP receiver will be responsible for translating the protocol extension to the
+existing protocol. Similarly, a new exporter will be responsible for translating the OTLP messages into this new Arrow-based
```

Comment:
Can we name the Collector receiver `otelarrow` and the exporter `otelarrow`?

Reply:
Sounds reasonable to me. I will discuss this with @jmacd  as he is managing this part of the system.

Reply:
Done